jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -1740,9 +1742,26 @@ public void setMetaData(MetaData.Request request)
                 throw new BadMessageException(badMessage);
         }
 
+        HttpField host = getHttpFields().getField(HttpHeader.HOST);
         if (uri.isAbsolute() && uri.hasAuthority() && uri.getPath() != null)
         {
             _uri = uri;
+            if (host instanceof HostPortHttpField && !((HostPortHttpField)host).getHostPort().toString().equals(uri.getAuthority()))
+            {
+                HttpConfiguration httpConfiguration = getHttpChannel().getHttpConfiguration();
+                if (httpConfiguration != null)
+                {
+                    if (httpConfiguration.getHttpCompliance().allows(MISMATCHED_AUTHORITY))
+                    {
+                        if (getHttpChannel() instanceof ComplianceViolation.Listener)
+                            ((ComplianceViolation.Listener)getHttpChannel()).onComplianceViolation(httpConfiguration.getHttpCompliance(), MISMATCHED_AUTHORITY, _uri.toString());
+                    }
```

Comment:
Please introduce variables for `HttpChannel`, `HttpCompliance` to make the code more readable?